src/inmanta/data/__init__.py

```diff
@@ -5578,29 +5579,25 @@ async def delete_cascade(self) -> None:
                 # Delete all code associated with this version
                 await Code.delete_all(connection=con, environment=self.environment, version=self.version)
 
-                # Acquire explicit lock to avoid deadlock. See ConfigurationModel docstring
-                await ResourceAction.lock_table(TableLockMode.ROW_EXCLUSIVE, connection=con)
-                await Resource.delete_all(connection=con, environment=self.environment, model=self.version)
+                # Delete ConfigurationModel and cascade delete on connected tables
+                await self.delete(connection=con)
```

Comment:
This is the main fix, the rest are some additional easy wins.

Reply:
Regarding 2: By splitting this into two different transactions we introduce a database leak, right? When the second query would be executed for some reason, the Parameter records will remain in the db until the end of times.